packages/mui-docs/src/MarkdownElement/MarkdownElement.tsx

```diff
@@ -823,21 +825,17 @@ export interface MarkdownElementProps {
 export const MarkdownElement = React.forwardRef<HTMLDivElement, MarkdownElementProps>(
   function MarkdownElement(props, ref) {
     const { className, renderedMarkdown, ...other } = props;
+    const rootRef = React.useRef<HTMLElement>(null);
+    const handleRef = useForkRef(rootRef, ref);
 
     React.useEffect(() => {
-      const elements = document.getElementsByClassName('title-link-to-anchor');
+      const elements = rootRef.current!.getElementsByClassName('title-link-to-anchor');
 
       for (let i = 0; i < elements.length; i += 1) {
         // More reliable than `-webkit-user-drag` (https://caniuse.com/webkit-user-drag)
         elements[i].setAttribute('draggable', 'false');
-        elements[i].addEventListener('click', handleClick, false);
+        elements[i].addEventListener('click', handleHeaderClick);
       }
-
-      return () => {
-        for (let i = 0; i < elements.length; i += 1) {
-          elements[i].removeEventListener('click', handleClick);
-        }
-      };
```

Comment:
If the component unmounts, these DOM nodes are removed. I imagine we don't need to remove event handlers.